ONSdigital/sdc-global-design-patterns
# 2568 input font sizes widths and button sizes

### What is the context of this PR?
* Create input/field width classes
* Ensure select is same height as text inputs
* Ensure buttons are same height as inputs for use in inline groups
* Change focus / checked styling to be driven by CSS rather than JS
* Make date input more mobile friendly

### How to review
* Check width classes make sense
* Ensure focused/checked styling is still applied correctly
* Review date input and ideally test with users

### Deploy preview
https://deploy-preview-204--sdc-global-design-patterns.netlify.com/

### Breaking changes
* Select placeholders (e.g. 'Please select') are now display none in option lists: https://github.com/ONSdigital/sdc-global-design-patterns/pull/204/files#diff-1562adaa907de8c36fc8a24c4c800c20R45
* Mobile base font size now matches desktop: https://github.com/ONSdigital/sdc-global-design-patterns/pull/204/files#diff-3314b0e9ef2da664ffc99f2b8aeb4910L8
* Input padding has been dropped from 0.6rem to 0.5rem https://github.com/ONSdigital/sdc-global-design-patterns/pull/204/files#diff-413036b74c400a478d1681913c21f0bfR2
* Input types (currency, unit, duration, etc.) no longer have specialised classes. Use just `input-type` and an `input-type--prefix` modifier for units such as currency: https://github.com/ONSdigital/sdc-global-design-patterns/pull/204/files#diff-a4f0a22443971354cb8e63aed8d5ff5b
* Fieldgroups no longer have specialised classes (`fieldgroup--date`, `fieldgroup--duration`). They now share a generic `fieldgroup--inline` class: https://github.com/ONSdigital/sdc-global-design-patterns/pull/204/files#diff-d71a01d2df8e281ad4281129ca8b81df
* Input focus/checked styles are now handled by CSS rather than JavaScript adding/removing classes: https://github.com/ONSdigital/sdc-global-design-patterns/pull/204/files#diff-2820a5e92afcecbe2f6697dee3cde559R100
